runtime/src/accounts_background_service.rs

```diff
@@ -122,8 +122,11 @@ impl SnapshotRequestHandler {
 
                 let mut hash_time = Measure::start("hash_time");
                 let mut hash_for_testing = None;
-                snapshot_root_bank
-                    .update_accounts_hash_with_index_option(true, test_hash_calculation);
+                const DO_NOT_USE_INDEX: bool = false;
+                snapshot_root_bank.update_accounts_hash_with_index_option(
+                    DO_NOT_USE_INDEX,
+                    test_hash_calculation,
+                );
```

Comment:
unrelated, but we could write hash_for_testing as 
```
let hash_for_testing = if test_hash_calculations {
     Some(..)
} else {
     None
 };
 ```

Reply:
handled in #15191